### PR TITLE
chore: bump ThreeJS to r128

### PR DESCRIPTION
**Pull request by larsmoa**

### CODE REVIEW

**Review by maksnester — CHANGES_REQUESTED** [review]

File: viewer/package.json [hunk 1]
```diff
@@ -90,21 +91,23 @@
...
   },
   "dependencies": {
-    "@cognite/potree-core": "^1.1.6",
+    "@cognite/potree-core": "1.2.0",
     "@cognite/reveal-parser-worker": "1.1.1",
     "@cognite/sdk-core": "^2.1.2",
     "@tweenjs/tween.js": "^18.6.4",
+    "@types/jsdom": "^16.2.10",
     "@types/lodash": "^4.14.170",
     "@types/skmeans": "^0.11.0",
```
Comment (maksnester) [inline viewer/package.json:100]: actually whatever that doesn't leak into public API must not be in dependencies. I don't think we provide lodash API anywhere?

**Comment by christjt** [comment]
1 Visual test i barely braking ~0.6% diff when we only allow 0.5% diff max. For now I suggest just bumping the threshold to allow it. For the future we should disable all post processing (FXAA, edges and SSAO) as a default for visual tests and have separate ones that cover these specific passes.

File: examples/package.json [hunk 2]
```diff
@@ -55,6 +55,7 @@
     "@types/react-dom": "^16.9.0",
     "@types/react-router-dom": "^5.1.6",
     "@types/styled-components": "^5.1.4",
+    "@types/three": "link:../viewer/node_modules/@types/three",
```
[the remark this answers is not in the document]
Comment (larsmoa) [reply examples/package.json:58]: Hm - it's a bit two-sided. Clients will actually often use ThreeJS, so it makes sense that it's here imo... Not sure how its best handled

**Comment by larsmoa** [comment]
> 1 Visual test i barely braking ~0.6% diff when we only allow 0.5% diff max. For now I suggest just bumping the threshold (like 0.8% or something) to allow it. For the future we should disable all post processing (FXAA, edges and SSAO) as a default for visual tests and have separate ones that cover these specific passes.

I will just update snapshot now and see if it stabilizes

File: viewer/package.json [hunk 3]
```diff
@@ -51,6 +51,7 @@
     "@types/jest": "^26.0.15",
     "@types/mixpanel-browser": "^2.35.4",
     "@types/node": "^14.14.5",
+    "@types/three": "0.128.0",
```
[the remark this answers is not in the document]
Comment (larsmoa) [reply viewer/package.json:57]: Moved - all of these are caused by me doing `yarn add @types/[...] --save-dev` which causes deps to be regular deps 🤦

File: examples/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (christjt) [reply examples/package.json:58]: I interperate this as you wouldn't link threejs like this in client apps. Rather import three js directly

[the remark this answers is not in the document]
Comment (larsmoa) [reply examples/package.json:58]: So we should rather keep `@types/three` in sync in viewer/ and examples/? What would that give us? Does it make sense to have @types/three be a common dependency so types always are available, just like three?

[the remark this answers is not in the document]
Comment (christjt) [reply examples/package.json:58]: It doesn't give us anything in practice, but at least it mimics our clients' environment. It doesn't matter that much to me though tbh. What do you think @Maksim?

[the remark this answers is not in the document]
Comment (larsmoa) [reply examples/package.json:58]: Maybe my suggestion about including @types/three is the right way to go about this?

[the remark this answers is not in the document]
Comment (maksnester) [reply examples/package.json:58]: > Clients will actually often use ThreeJS

I think the safest way for clients here will be to use THREE from Reveal. Otherwise, there always a chance for some collisions, no matter how to configure three in reveal (as a peer or as a common dependency).

> so it makes sense that it's here imo

I agree it makes sense to have some of our environments closer to clients setup, maybe we should check what will happen for three.js versions mismatch in our examples. But I guess that 0 clients have a link on the exact three.js version that Reveal uses in their package.json. 

> we should rather keep @types/three in sync in viewer/ and examples/

Hm... I would say we shouldn't have them linked at least. Linking is a cheat to have them in sync, but speaking about clients, they probably don't want to sync anything with Reveal internals. So by not doing syncs we at least might discover when they are actually must be done (or things won't work). 

TLDR; I suggest to avoid npm-links from Reveal node_modules in examples & documentations to have a better understanding of what will go wrong in clients projects. So at least we will know whether that sync will be necessary and when. 

@larsmoa @christjt

[the remark this answers is not in the document]
Comment (larsmoa) [reply examples/package.json:58]: Ok, if we include types/three in Reveal we can remove the links, right? I suggest we do that

**Review by maksnester — APPROVED** [review]
LGTM

File: examples/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (maksnester) [reply examples/package.json]: > if we include types/three in Reveal we can remove the links, right

@larsmoa it depends... If we use THREE only from reveal as `import { THREE } from '@cognite/reveal` then yes, we can remove the links and that's it.

If we (or client app) want to import things from `three` directly, then `three` and `@types/three` still need to be installed. In some cases, Reveal and a client app will have the same version of three installed, and sometimes they won't. And when they won't, in theory, there might be compile-time issues with three.js types when they are not compatible
